game-core/src/main/java/games/strategy/engine/data/changefactory/AddUnits.java

```diff
@@ -15,19 +20,26 @@
   private final String name;
   private final Collection<Unit> units;
   private final String type;
+  private Map<UUID, String> unitPlayerMap;
 
   AddUnits(final UnitCollection collection, final Collection<Unit> units) {
-    this.units = new ArrayList<>(units);
+    this.units = units;
+    unitPlayerMap = getUnitPlayerMap(units);
     name = collection.getHolder().getName();
     type = collection.getHolder().getType();
   }
 
   AddUnits(final String name, final String type, final Collection<Unit> units) {
-    this.units = new ArrayList<>(units);
+    this.units = units;
+    unitPlayerMap = getUnitPlayerMap(units);
     this.type = type;
     this.name = name;
   }
 
+  private Map<UUID, String> getUnitPlayerMap(final Collection<Unit> units) {
```

Comment:
nit: 'get' methods usually are just 'getters'. If computation is done a different verb could be easier for future maintainers. Perhaps rewording this to `buildUnitPlayerMap` would help that. WDYT @trevan ?

Reply:
This could be a moot comment if it turns out we do not need a map.